Enable HLS video encodings for mobile.

waheedahmed opened the pull request:
Return HLS video encodings from blocks api for mobile.

LEARNER-4167

waheedahmed commented:
jenkins run a11y

waheedahmed commented:
@edx/learner-spartans Please review.

Qubad786 commented on common/lib/xmodule/xmodule/tests/test_video.py:
```diff
@@ -883,6 +883,32 @@ def test_student_view_data(self, field_data, expected_student_view_data, mock_tr
+        descriptor.runtime.course_id = MagicMock()
+        student_view_data = descriptor.student_view_data()
+        expected_video_data = {u'hls': {'url': u'http://www.meowmix.com', 'file_size': 25556}}
+        self.assertEquals(student_view_data.get('encoded_videos'), expected_video_data)
```
`self.assertEquals` is a deprecated alias: https://docs.python.org/2/library/unittest.html#deprecated-aliases, `self.assertEqual` can replace it but in our case, we should really be using `self.assertDictEqual`.

Qubad786 commented on common/lib/xmodule/xmodule/tests/test_video.py:
```diff
@@ -883,6 +883,32 @@ def test_student_view_data(self, field_data, expected_student_view_data, mock_tr
+            ]
+        }
+
+        descriptor = instantiate_descriptor(**{'edx_video_id': 'example_id', 'only_on_web': False})
```
Better if we do as:
```bash
instantiate_descriptor(edx_video_id='example_id', only_on_web=False)
```

Qubad786 commented:
@waheedahmed is there any sandbox to test this?

waheedahmed commented:
@Qubad786 Addressed your comments and here is the [sandbox link](https://hls-videos.sandbox.edx.org/api/courses/v1/blocks/?depth=all&requested_fields=graded,format,student_view_multi_device,due&student_view_data=video,discussion&block_counts=video&nav_depth=3&username=edx&course_id=course-v1%3AedX%2BDemoX%2BDemo_Course) for testing. I have configured the HLS video for demo course and blocks api returning it successfully.
